### PR TITLE
clampBits() changes

### PR DESCRIPTION
**Pull request by axxel**
Fix most of the remaining core library issues (excluding Test and utilities).

The fix of the `fixBadPixel` copy'n'paste bug (last commit) results in a bunch of rstest hash mismatches.

### CODE REVIEW

File: src/librawspeed/common/Common.h [hunk 1]
```diff
@@ -26,10 +26,7 @@
 #include <memory>           // for unique_ptr, allocator
 #include <string>           // for string
 #include <vector>           // for vector
-
-#ifndef __BYTE_ORDER__
-#include <cassert> // for assert
```
Comment (LebedevRI) [inline src/librawspeed/common/Common.h]: If you don't know why something is there, maybe not change it

File: src/librawspeed/decoders/CrwDecoder.cpp [hunk 2]
```diff
@@ -193,8 +193,9 @@ void CrwDecoder::decodeMetaDataInternal(CameraMetaData *meta) {
       ushort16 wb_index = shot_info->getU16(7);
       CiffEntry *wb_data = mRootIFD->getEntryRecursive(CIFF_WHITEBALANCE);
       /* CANON EOS D60, CANON EOS 10D, CANON EOS 300D */
-      int wb_offset = (wb_index < 18) ? "0134567028"[wb_index]-'0' : 0;
-      wb_offset = 1+wb_offset*4;
+      if (wb_index > 9)
+        ThrowIOE("CrwDecoder: invalid white balance index");
```
Comment (LebedevRI) [inline src/librawspeed/decoders/CrwDecoder.cpp:197]: how is this an input/output issue?

File: src/librawspeed/common/Common.h [hunk 3]
```diff
@@ -107,10 +108,11 @@ extern "C" { FILE __iob_func[3] = { *stdin,*stdout,*stderr }; }
 #endif
 #endif
 
-// clampBits clamps the given int to the range 0 .. 2^n-1
-inline uint32 clampBits(int64 x, uint32 n) {
-  const int64 tmp = (1ULL << n) - 1ULL;
-  x = x < 0LL ? 0LL : x;
+// clampBits clamps the given int to the range 0 .. 2^n-1, with n <= 16
+inline ushort16 clampBits(int x, uint32 n) {
+  assert(n <= 16);
```
Comment (LebedevRI) [inline src/librawspeed/common/Common.h:113]: So what about dngopcodes where it is probably used with wrong argument order?

**Comment by LebedevRI** [comment]
Picked all but `clampBits()` commit, thanks.
I don't agree with UNINIT_CTOR fixes, they only silence the warning, the real issues are still there.

File: src/librawspeed/common/Common.h (hunk 3 above)
Comment (axxel) [reply src/librawspeed/common/Common.h:113]: I fixed those in #80.

File: src/librawspeed/decoders/CrwDecoder.cpp (hunk 2 above)
Comment (axxel) [reply src/librawspeed/decoders/CrwDecoder.cpp:197]: The type of the exception might be questionable (as are _all_ exception types). The fact is, if `wb_index > 9` then the file is broken.

Comment (LebedevRI) [reply src/librawspeed/decoders/CrwDecoder.cpp:197]: True

**Pull request closed without merging**